### DISCUSSION TITLE
Can I use it in jupyter? Has It a jupyter kernnel?

### DISCUSSION THREAD

**Discussion by baiguoname** [sigmaSd/IRust#107] [General]
Jupyter is amazing, can I use Irust in Jupyter, especially in vscode-jupyter.

**Comment by sigmaSd** [comment]
Hello, it's possible https://github.com/sigmaSd/IRust/tree/master/crates/irust_repl#jupyter-kernel but doesn't currently have first class support

https://github.com/google/evcxr has support for this

**Reply by baiguoname** [reply]
I am using evcxr, but I found there is many limitation in that for now, such as a local file can not be imported into. I checked the :help in Irust, and I found many feature I may use often. I would like to know if irust for jupyter just has the same features as the original irust? What is the limitation to it both compared to Irust and evcxr?

**Reply by sigmaSd** [reply]
The current present kernel https://github.com/sigmaSd/IRust/blob/master/crates/irust_repl/irustkernel/irust.py uses only rust-repl crate and in the simplest way possible.

You can try that and see if does what you want. To see what you can do with rust repl search for `pub fn` in https://github.com/sigmaSd/IRust/blob/master/crates/irust_repl/src/lib.rs

Some of the irust functionality is implemented outside of it in the parser so those parts wont be usable, or would need to  be to added to the current kernel https://github.com/sigmaSd/IRust/blob/master/crates/irust/src/irust/parser.rs#L48 

What would improve things is if irust was usable as child process (which will allow the kernel to use it) but that's currently blocked on https://github.com/crossterm-rs/crossterm/pull/711 so either that gets fixed or this https://github.com/crossterm-rs/crossterm/pull/711

TLDR: you can try https://github.com/sigmaSd/IRust/blob/master/crates/irust_repl/irustkernel/ it will probably need some hacking on  to do what you expect, in the future things can be better if the kernel can use irust directly
